Sources/SwiftKafka/KafkaClient.swift

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import Logging
+
+/// Base class for ``KafkaProducer`` and ``KafkaConsumer``, which is used to handle the connection to the Kafka ecosystem.
+class KafkaClient {
+    // Default size for Strings returned from C API
+    static let stringSize = 1024
+
+    let logger: Logger
+
+    private let _clientType: rd_kafka_type_t
+    private let _config: KafkaConfig
+    private let _kafkaHandle: OpaquePointer
```

Comment:
I think we can drop the `_` here and could we add some doc comments what each of these are.